### ISSUE TITLE
Review FlyClient proposal for possible impact on Light client direction

### ISSUE THREAD

**Issue by FrankSzendzielarz** [ethereum/devp2p#86]
There is a "FlyClient" proposal that seems to offer an alternative approach for constrained resource clients . The document is at the [this location]( https://eprint.iacr.org/2019/226.pdf) .

A review would help us understand

- What it offers
- Should it be implemented and if so how
- What is the effect on the converged light protocols project (bringing Parity and geth in alignment for the future of PIP + LES)

**Comment by rjl493456442** [comment]
Never found it until now! Actually I read this paper several months ago. 

TL;DR the main idea of this paper is: the client can verify a checkpoint of blockchain without the trust of any third party. 

So that when a  client joins the network, it can:
* request a checkpoint of longest blockchain;
* verify the validity of checkpoint via sample(I think it uses the similar idea of zk, based on the formula the paper constructs, the client can detect fraud with a very high chance in limited steps.);
* sync the remaining header based on the verified checkpoint;

Personally I think it's awesome. Now we use a centralized way to prove the checkpoint is valid(only foundation developers can generate checkpoints). A trustless checkpoint is always the goal we pursue
